src/Channels/Channel.cs

```diff
@@ -340,14 +350,15 @@ public void AdvanceReader(ReadCursor consumed, ReadCursor examined)
             PooledBufferSegment returnStart = null;
             PooledBufferSegment returnEnd = null;
 
+            // Reading commit head shared with writer
```

Comment:
Resolved, IsEnd was wrong thing to be checking anyway as would compare with writer end